### PR TITLE
Add .gitignore to .azure folder to make sure it is not committed

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-dev/issues/2755

### CODE REVIEW

File: cli/azd/pkg/environment/azdcontext/azdcontext.go [hunk 1]
```diff
@@ -68,14 +68,26 @@ func (c *AzdContext) GetDefaultEnvironmentName() (string, error) {
...
+	}
+
+	if err := writeConfig(path, config); err != nil {
+		return err
 	}
 
-	return writeConfig(path, config)
+	// make sure to ignore the environment directory
+	path = filepath.Join(c.EnvironmentDirectory(), ".gitignore")
+	return os.WriteFile(path, []byte("# .azure is not intended to be committed\n*"), osutil.PermissionFile)
```
Comment (ellismg) [inline cli/azd/pkg/environment/azdcontext/azdcontext.go:90]: Should we do this write only if `.gitignore` doesn't already exist, to allow the user to change the contents if they wished?

**Comment by ellismg** [comment]
Would we want to stop writing `.azure` into the `.gitignore` next to `.azure` now?

File: cli/azd/pkg/environment/azdcontext/azdcontext.go (hunk 1 above)
Comment (rajeshkamal5050) [reply cli/azd/pkg/environment/azdcontext/azdcontext.go:90]: Could we also append .azure, if `.gitignore` already exists?

Comment (wbreza) [reply cli/azd/pkg/environment/azdcontext/azdcontext.go:90]: > Should we do this write only if `.gitignore` doesn't already exist, to allow the user to change the contents if they wished?

This seems to be the safest thing to do - and if they really wanted to commit something they could add perform a force add to override the .gitignore.

**Review by wbreza — APPROVED** [review]

**Comment by jongio** [comment]
Does this?

1. On every azd command:
1. Check for .azure/.gitignore
1. If not there create it
1. If there and doesn't have .azure, then add it?

**Comment by vhvb1989** [comment]
> Does this?
> 
> 1. On every azd command:
> 2. Check for .azure/.gitignore
> 3. If not there create it
> 4. If there and doesn't have .azure, then add it?

No.  There's no `check for .azure/.gitignore`.  Azd creates/saves a `.azure/.gitignore` file every time an environment is saved to disk (.env + config.json).  The content of the `.azure/.gitignore` file looks like:

```
# .azure is not intended to be committed to the repository
*
```

The `*` makes git to ignore ALL the contents from that folder (.azure).


Then:
- If you run `azd env new`, the `.azure/.gitignore` file is saved/created (regardless if it already exists).
- If you run `azd provision`, the `.azure/.gitignore` file is saved/created as well after writing the bicep outputs to the .env
- Same for `azd env set foo bar`.

And:
- If a user manually changes the `.azure/.gitignore` file (like change the * to just ignore just some environments), the change is lost/overridden by azd.  This makes hard to actually commit anything from within the folder.  If you want to do that, you would manually need to make the change and then manually run `git commit` and `git push`.

- Even when running `azd pipeline config`, azd will first save/create the gitignore, preventing itself from ever committing the content from that folder.

**Comment by jongio** [comment]
> > Does this?
> > 
> > 1. On every azd command:
> > 2. Check for .azure/.gitignore
> > 3. If not there create it
> > 4. If there and doesn't have .azure, then add it?
> 
> No. There's no `check for .azure/.gitignore`. Azd creates/saves a `.azure/.gitignore` file every time an environment is saved to disk (.env + config.json). The content of the `.azure/.gitignore` file looks like:
> 
> ```
> # .azure is not intended to be committed to the repository
> *
> ```
> 
> The `*` makes git to ignore ALL the contents from that folder (.azure).
> 
> Then:
> 
> * If you run `azd env new`, the `.azure/.gitignore` file is saved/created (regardless if it already exists).
> * If you run `azd provision`, the `.azure/.gitignore` file is saved/created as well after writing the bicep outputs to the .env
> * Same for `azd env set foo bar`.
> 
> And:
> 
> * If a user manually changes the `.azure/.gitignore` file (like change the * to just ignore just some environments), the change is lost/overridden by azd.  This makes hard to actually commit anything from within the folder.  If you want to do that, you would manually need to make the change and then manually run `git commit` and `git push`.
> * Even when running `azd pipeline config`, azd will first save/create the gitignore, preventing itself from ever committing the content from that folder.

Will it verify it is there when a user saves a env value with `azd env set`?

**Pull request merged**